lib/c-api/src/wasm_c_api/wat.rs

```diff
@@ -0,0 +1,28 @@
+use super::types::wasm_byte_vec_t;
+
+/// Parses in-memory bytes as either the WAT format, or a binary Wasm
+/// module. This is wasmer-specific.
+///
+/// In case of failure, `wat2wasm` returns `NULL`.
+#[cfg(feature = "wat")]
+#[no_mangle]
+pub unsafe extern "C" fn wat2wasm(wat: &wasm_byte_vec_t) -> Option<Box<wasm_byte_vec_t>> {
+    let wat: &[u8] = wat.into_slice()?;
+
+    let result = match wasmer::wat2wasm(wat) {
+        Ok(result) => result,
+        Err(error) => {
+            crate::error::update_last_error(error);
+
+            return None;
+        }
+    };
```

Comment:
Is this different than what `c_try!` does?

Reply:
It was, but now that the return type is an `Option`, it's not!